src/libstd/sys/windows/args.rs

```diff
@@ -12,32 +12,76 @@
 
 use os::windows::prelude::*;
 use sys::c;
-use slice;
-use ops::Range;
 use ffi::OsString;
-use libc::{c_int, c_void};
 use fmt;
+use collections::VecDeque;
+use core::iter;
 
 pub unsafe fn init(_argc: isize, _argv: *const *const u8) { }
 
 pub unsafe fn cleanup() { }
 
 pub fn args() -> Args {
     unsafe {
-        let mut nArgs: c_int = 0;
-        let lpCmdLine = c::GetCommandLineW();
-        let szArgList = c::CommandLineToArgvW(lpCmdLine, &mut nArgs);
-
-        // szArcList can be NULL if CommandLinToArgvW failed,
-        // but in that case nArgs is 0 so we won't actually
-        // try to read a null pointer
-        Args { cur: szArgList, range: 0..(nArgs as isize) }
+        let lp_cmd_line = c::GetCommandLineW();
+        let parsed_args_list = parse_lp_cmd_line(lp_cmd_line as *const u16);
+        Args { parsed_args_list: parsed_args_list }
     }
 }
 
+/// https://docs.microsoft.com/en-us/previous-versions//17w5ykft(v=vs.85)
+pub unsafe fn parse_lp_cmd_line(lp_cmd_line: *const u16) -> VecDeque<OsString> {
+    let mut in_quotes = false;
+    let mut backslash_count: usize = 0;
+    let mut i = 0;
+    let mut ret_val = VecDeque::new();
+    let mut cur = Vec::new();
+    loop {
+        let c = lp_cmd_line.offset(i);
+        match *c {
+            // backslash
+            0x5c => backslash_count += 1,
+            // double quote
+            0x22 if backslash_count % 2 == 0 => {
+                cur.extend(iter::repeat(b'\\' as u16).take(backslash_count / 2));
+                backslash_count = 0;
+                in_quotes = !in_quotes;
+            }
+            0x22 if backslash_count % 2 != 0 => {
```

Comment:
Could 0x22 here (and in other locations) be listed as `'"' as u16` perhaps?

Reply:
'\\' as u16 => backslash_count += 1,
         ^^ expected one of `...`, `..=`, `..`, `=>`, `if`, or `|` here

Reply:
Could... something in the spirit of the comment be tried? Can parentheses be placed around the pattern? Could the pattern be extracted to a static? (etc)